### ISSUE TITLE
--depth overrides default for --number-of-lines

### ISSUE THREAD

**Issue by asomers** [bootandy/dust#312]
By default, dust chooses a good value for `--number-of-lines`.  But if I also use `--depth`, then dust acts as though `--number-of-lines` is unlimited, and prints everything.  Shouldn't it still use a reasonable number?

FreeBSD 14.0-CURRENT amd64, Dust 0.8.1.

**Comment by bootandy** [comment]
I decided that using` --depth `meant the user wanted to go see everything to that level. 

If `--depth` assumed the default number of lines of output it could lead to strange behaviour. 

* If you choose `--depth=1` and there is one directory and two files. Should it only show 3 entries or should it open the directory and go into lower depths too?

* If you select `depth=3` and all the big files are at depth=1 then does it only show the big files at depth=1 and not go down to show you the directory structure of the nested directories?

I wrestled with the above ideas and thought that if you want a correct `--number-of-lines` you probably don't want to use `--depth` at all. Just run dust normally.
